src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,419 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DiaSymReader.Tools;
+using Microsoft.Tools.Common;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml;
+
+namespace Microsoft.Diagnostics.Tools.Stack
+{
+    internal static class SymbolicateHandler
+    {
+        // Temporary folder to store the files converted from pdb to xml
+        private static readonly string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+        delegate void SymbolicateDelegate(FileInfo[] inputPath, string searchDir, string output);
+
+        /// <summary>
+        /// Get the line number from the Method Token and IL Offset at the stacktrace
+        /// </summary>
+        /// <param name="inputPath">The input path for file with stacktrace text</param>
+        /// <param name="searchDir">All paths in the directory to the assembly and pdb where the exception occurred</param>
+        /// <param name="output">The output path for the extracted line number data</param>
+        /// <returns></returns>
+        private static void Symbolicate(FileInfo[] inputPath, string searchDir, string output)
+        {
+            try
+            {
+                if (inputPath == null)
+                {
+                    throw new InvalidDataException("Required argument missing for command\n");
+                }
+
+                string input = string.Empty;
+                if (inputPath.Length > 1)
+                {
+                    for (int i = 0; i < inputPath.Length; i++)
+                    {
+                        Console.WriteLine($" {i + 1}. {inputPath[i]}");
```

Comment:
Rather than calling the static `Console` you can add a parameter of type `IConsole` to the method. This also makes testing a lot easier as you can pass in a `TestConsole` and capture its output in your tests, avoiding side effects between tests that you'll see when calling the static `Console` methods.

Reply:
I modified it.